src/date-picker/date-picker.jsx

```diff
@@ -113,18 +116,38 @@ let DatePicker = React.createClass({
     return this.state.date;
   },
 
+  /**
+   * Setting the state will update the date here and also re-render the TextField with
+   * the new value.
+   */
   setDate(d) {
     this.setState({
       date: d,
     });
```

Comment:
Should there be an error here to not allow `setDate` if the component is controlled?